File: Cpp/Medium/MinimumSizeSubarraySum.cpp
/**********************************************************
 *
 *  209. Minimum Size Subarray Sum solution c++
 *
 *  Joel M. Gonzalez Rodriguez
 *
*/*********************************************************

int minSubArrayLen(int target, vector<int> &nums)
{
    int l, r, ans, sum;
    l = r = sum = 0;
    ans = INT_MAX;

    while (r < nums.size())
    {
        sum += nums[r];
        while (sum >= target)
        {
            ans = min(ans, (r - l) + 1);
            sum -= nums[l];
            l++;
        }
        r++;
    }

    if (ans == INT_MAX)
        return 0;

    else
        return ans;
}

File: Python/Medium/Minimum Size_Subarray_Sum.py
'''
    209. Minimum Size Subarray Sum

    Solution:
'''

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l, r, curr_sum = 0, 0
        res = float('inf')

        while r < len(nums):
            curr_sum += nums[r]
            while curr_sum >= target:
                res = min(res, r - l + 1)
                curr_sum -= nums[l]
                l += 1

            r += 1

        return res if res != float('inf') else 0